source/base/mpi.cc

```diff
@@ -528,30 +528,46 @@ namespace Utilities
 
     MinMaxAvg
     min_max_avg(const double my_value, const MPI_Comm &mpi_communicator)
+    {
+      return min_max_avg(std::vector<double>{my_value}, mpi_communicator)[0];
+    }
+
+    std::vector<MinMaxAvg>
+    min_max_avg(const std::vector<double> my_values,
+                const MPI_Comm &          mpi_communicator)
     {
       // If MPI was not started, we have a serial computation and cannot run
       // the other MPI commands
       if (job_supports_mpi() == false)
         {
-          MinMaxAvg result;
-          result.sum       = my_value;
-          result.avg       = my_value;
-          result.min       = my_value;
-          result.max       = my_value;
-          result.min_index = 0;
-          result.max_index = 0;
+          std::vector<MinMaxAvg> result(my_values.size());
+
+          for (unsigned int i = 0; i < my_values.size(); i++)
+            {
+              result[i].sum       = my_values[i];
+              result[i].avg       = my_values[i];
+              result[i].min       = my_values[i];
+              result[i].max       = my_values[i];
+              result[i].min_index = 0;
+              result[i].max_index = 0;
```

Comment:
Even though I am normally against adding new memory allocations for cheap operations like the non-parallel case here (note that I added an additional switch in #9472 to avoid creating the MPI data structures, which showed up in my profiles due to several memory allocations), due to the low frequency by which we call `min_max_avg` I think it is indeed ok here. Well, the only case I can imagine it would hurt is if we have `Timer::stop()` inside an inner loop (taking less than 1us), and in that case one should not use the standard timer classes anyway.